File: src/omichub/application/schemas/announcement.py
"""首页条幅通知 DTO"""

from datetime import datetime
from uuid import UUID

from omichub.application.schemas.base import OmicsHubBaseSchema


class AnnouncementCreate(OmicsHubBaseSchema):
    """创建通知请求"""

    title: str
    description: str
    type: str = "info"
    icon: str | None = None
    link: str | None = None
    button_text: str | None = None
    start_time: datetime
    end_time: datetime | None = None
    dismiss_behavior: str = "none"
    priority: int = 0
    is_enabled: bool = True


class AnnouncementUpdate(OmicsHubBaseSchema):
    """更新通知请求（全字段可选，仅传需更新的字段）"""

    title: str | None = None
    description: str | None = None
    type: str | None = None
    icon: str | None = None
    link: str | None = None
    button_text: str | None = None
    start_time: datetime | None = None
    end_time: datetime | None = None
    dismiss_behavior: str | None = None
    priority: int | None = None
    is_enabled: bool | None = None


class AnnouncementResponse(OmicsHubBaseSchema):
    """通知响应"""

    id: UUID
    title: str
    description: str
    type: str
    icon: str | None = None
    link: str | None = None
    button_text: str | None = None
    start_time: datetime
    end_time: datetime | None = None
    dismiss_behavior: str
    priority: int
    is_enabled: bool
    created_at: datetime
    updated_at: datetime

File: frontend/src/types/announcement.ts
/** 首页条幅通知类型定义（字段与后端 AnnouncementResponse 对齐，snake_case） */

export type AnnouncementType = 'info' | 'success' | 'warning' | 'feature'

export type DismissBehavior = 'daily' | 'forever' | 'none'

export interface Announcement {
  id: string
  title: string
  description: string
  type: AnnouncementType
  icon?: string | null
  link?: string | null
  button_text?: string | null
  start_time: string
  end_time?: string | null
  dismiss_behavior: DismissBehavior
  priority: number
  is_enabled: boolean
  created_at: string
  updated_at: string
}

/** 创建/更新请求体（与后端 AnnouncementCreate 对齐） */
export type AnnouncementPayload = Omit<
  Announcement,
  'id' | 'created_at' | 'updated_at'
>

/** 类型 → 默认图标映射（管理员未设 icon 时用） */
export const DEFAULT_ICON_BY_TYPE: Record<AnnouncementType, string> = {
  info: '📢',
  success: '🎉',
  warning: '⚠️',
  feature: '✨',
}
